### ISSUE TITLE
Remove transpose before reshape

### ISSUE THREAD

**Issue by H4dr1en** [PINTO0309/onnx2tf#193]
### Issue Type

Feature Request

### onnx2tf version number

1.6.6

### onnx version number

1.13.0

### tensorflow version number

2.10.0

### Download URL for ONNX

model without transpose: https://drive.google.com/file/d/1cOIdoZJ1qS-pvVrihk1EumT6tQMI24PD/view?usp=sharing

With extra last dim: https://drive.google.com/file/d/1uT9xqnY7iY9ML7OIaaouETuLQXkdMmCV/view?usp=sharing

### Parameter Replacement JSON

```json
{}
```

### Description

1. Product Development. Fixing this bug will allow me to run the model on prod
2. I am running tflite models on android devices without openCL and the app crashes with the error that the transpose layer cannot be run on GPU, so it falls back to openCL implementation and crashes because this library is not available.
Long story short, I want to get rid of the single transpose layer introduced by the conversion from onnx to tflite.

| Onnx model without transpose  | Tflite model with transpose |
| ------------- | ------------- |
| <img width="262" alt="Screenshot 2023-02-17 at 14 59 28" src="https://user-images.githubusercontent.com/16240134/219677277-244ceffb-be9c-4720-9a34-b553fc699924.png">  | <img width="253" alt="Screenshot 2023-02-17 at 14 59 42" src="https://user-images.githubusercontent.com/16240134/219677352-a935c293-2e99-4e86-b92f-640cd696669a.png">  |

Code for model:
```
class Model(torch.nn.Module):
    def forward(x):   # (B, C, H, W)
        flat = x.view(1, -1, 64 * 48)  # (B, C, H * W)
        flat = flat - torch.max(flat, dim=-1, keepdim=True).values
        flat = torch.nn.functional.softmax(flat, -1)
        return flat.sum(axis=-1)  # (B, C)
```
3. I tried various workarounds to prevent a transpose to be added, but could not succeed so far. It's not clear to my why onnx2tf doesn't do channel last and introduce a transpose layer. Is there a way to overcome that?

I tried to add an extra dummy dimension to fool the converter, but this confuses the converter that does the reshaping incorrectly:

```
class Model(torch.nn.Module):
    def forward(x):   # (B, C, H, W)
        flat = x[..., None]:   # (B, C, H, W, 1)
        flat = flat.view(1, -1, 64 * 48, 1)
        flat = flat - torch.max(flat, dim=-2, keepdim=True).values
        flat = torch.nn.functional.softmax(flat, -2)
        return flat[..., 0]
```

| Onnx model  | Tflite model |
| ------------- | ------------- |
| <img width="288" alt="Screenshot 2023-02-17 at 15 13 49" src="https://user-images.githubusercontent.com/16240134/219679095-723ce20b-5004-47d9-a3f6-22f026ecb463.png">  | <img width="256" alt="Screenshot 2023-02-17 at 15 15 49" src="https://user-images.githubusercontent.com/16240134/219679242-68727daa-950b-4095-b209-5048b3ac3674.png">  |

5. I need the model to run on these android devices

**Comment by PINTO0309** [comment]
```
onnx2tf -i model.onnx -kat image
```
![image](https://user-images.githubusercontent.com/33194443/219680921-eb555783-4a32-4aa5-a748-984f38fe0b47.png)

**Comment by PINTO0309** [comment]
Frankly, I don't understand the nature of the problem you are having because only part of the model has been shared. Even if there is a realistic solution, it is often impossible to answer if there is insufficient information. I am not an esper.

**Comment by H4dr1en** [comment]
Yes I understand, I'll update the model to include some convs and ping here

**Comment by H4dr1en** [comment]
Here is the updated model: https://drive.google.com/file/d/1_jGqloclOR-hfKToyAUm-3a4czsrxqcx/view?usp=sharing

It is still simplified (There are many more convs than one) but should better reflect the problem: The input is (1, 3, 456, 192) and after all the conv layers, output is (1, 23, 64, 48). At this point, onnx2tf introduces the transpose layer that I need to get rid of

| Onnx  | TFLITE |
| ------------- | ------------- |
| <img width="284" alt="Screenshot 2023-02-17 at 15 50 39" src="https://user-images.githubusercontent.com/16240134/219687458-4db22300-3b01-4ca9-b4ed-084308d03e4f.png">  | <img width="212" alt="Screenshot 2023-02-17 at 15 50 58" src="https://user-images.githubusercontent.com/16240134/219687494-77bee4d9-4c56-4366-93bd-9fd1ce627010.png">  |

**Comment by PINTO0309** [comment]
Just to be sure, is there no Transpose other than this part? I am referring to the ONNX model.

**Comment by H4dr1en** [comment]
Yes, no transpose in the ONNX model

I am using pytorch to develop the models, and on android devices, I am using the `org.tensorflow:tensorflow-lite` / `org.tensorflow:tensorflow-lite-gpu` modules to run the inference on the tflite model. It can run transpose, only not on all devices, such as nexus phones

**Comment by PINTO0309** [comment]
Your request is too device specific to be addressed by tool side behavior. Reshaping a tensor with NCW, NCHW, or NCDHW geometry will always break all models except yours unless you extrapolate the Transpose to NWC, NHWC, or NDHWC. In other words, if you want to remove Transpose by any means, just separate it into two dimensions or less.
```python
import torch

class Model(torch.nn.Module):
    def __init__(self):
        super(Model, self).__init__()

        self.conv = torch.nn.Conv2d(3, 23, (5,5), stride=4, padding=2)

    def forward(self, x):   # (B, C, H, W)
        y = self.conv(x)
        zs = torch.split(y, split_size_or_sections=1, dim=1)
        aaa = []
        for i in zs:
            bbb = torch.squeeze(i)
            ccc = bbb.reshape([3072])
            ddd = torch.unsqueeze(ccc, 0)
            eee = torch.unsqueeze(ddd, 0)
            aaa.append(eee)
        fff = torch.cat(aaa, dim=1)
        flat = fff - torch.max(fff, dim=-1, keepdim=True).values
        flat = torch.nn.functional.softmax(flat, -1)
        return flat.sum(axis=-1)

model = Model()

x = torch.randn([1,3,256,192])
onnx_file = f'aaa.onnx'
torch.onnx.export(
    model,
    args=(x),
    f=onnx_file,
    opset_version=11,
    input_names=[
        'input',
    ],
    output_names=[
        'output',
    ],
)
import onnx
from onnxsim import simplify
model_onnx2 = onnx.load(onnx_file)
model_simp, check = simplify(model_onnx2)
onnx.save(model_simp, onnx_file)
```
![image](https://user-images.githubusercontent.com/33194443/219705696-9477a013-6b4c-4a35-8013-e2f2eb063a61.png)

**Comment by PINTO0309** [comment]
```
onnx2tf -i model_2.onnx -nodafc 2
```
![image](https://user-images.githubusercontent.com/33194443/219749970-246fe412-a33a-43a7-af98-0d8cc966bcca.png)

https://github.com/PINTO0309/onnx2tf/releases/tag/1.7.1

**Comment by PINTO0309** [comment]
https://github.com/PINTO0309/onnx2tf/releases/tag/1.7.2
